### PR TITLE
HOMEWORK:  Valentin 25.10

### PR DESCRIPTION
**Pull request by alt-plus-f4**
files:
homeworks/HW-10-25-AsyncFileServer

### CODE REVIEW

File: homeworks/HW-10-25-AsyncFileServer/AsyncFileClient.java
Comment (alt-plus-f4) [inline homeworks/HW-10-25-AsyncFileServer/AsyncFileClient.java:1]: Client

File: homeworks/HW-10-25-AsyncFileServer/AsyncFileServer.java
Comment (alt-plus-f4) [inline homeworks/HW-10-25-AsyncFileServer/AsyncFileServer.java:1]: Server

**Review by alt-plus-f4 — COMMENTED** [review]
Marked files

File: homeworks/HW-10-25-AsyncFileServer/AsyncFileServer.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            } catch (IOException | InterruptedException | ExecutionException e) {
+                logger.log(Level.SEVERE, "Server encountered an error", e);
+            }
+        }
+    }
+
+    private void handleClient(AsynchronousSocketChannel client) throws IOException, InterruptedException, ExecutionException {
+        ByteBuffer buffer = ByteBuffer.allocate(1024);
+        
+        client.read(buffer).get();
```
Comment (tvuchova) [inline homeworks/HW-10-25-AsyncFileServer/AsyncFileServer.java:48]: no need of get here ;
  
   ```
   client.read(buffer, buffer, new CompletionHandler<Integer, ByteBuffer>() {
            @Override
            public void completed(Integer result, ByteBuffer attachment) {
                if (result == -1) {
                    closeClient(client);
                    return;
                }
                buffer.flip();
                String fileName = new String(buffer.array(), 0, result).trim();
                buffer.clear();
                receiveFile(client, fileName);
..........................................
            }
```

File: homeworks/HW-10-25-AsyncFileServer/AsyncFileClient.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            future.get();
+            System.out.println("Connected to server at " + host + ":" + port);
+        } catch (IOException | InterruptedException | ExecutionException e) {
+            logger.log(Level.SEVERE, "Connection error", e);
+        }
+        socketChannel = tempChannel;
+    }
+
+    public void uploadFile(String filePath) {
+        Path path = Paths.get(filePath);
```
Comment (tvuchova) [inline homeworks/HW-10-25-AsyncFileServer/AsyncFileClient.java:43]: if (!Files.exists(path) || !Files.isReadable(path)) {
            logger.log(Level.SEVERE, "File does not exist or is not readable: " + filePath);
            return;
        }

File: homeworks/HW-10-25-AsyncFileServer/HW-10-25-AsyncFileServer.iml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="JAVA_MODULE" version="4">
```
Comment (tvuchova) [inline homeworks/HW-10-25-AsyncFileServer/HW-10-25-AsyncFileServer.iml:2]: you can add .iml file and .idea in gitignore

**Pull request closed without merging**